### ISSUE TITLE
[BUG] Validation fails when anchor or alias a in the YAML file which seems correct

### ISSUE THREAD

**Issue by lfoppiano** [OpenAPITools/openapi-generator#11770]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

Basically the YAML is correct but the validation fails when anchors or aliases are present.  

##### openapi-generator version

5.4.0 from homebrew 

##### OpenAPI declaration file content or url

[spec.yaml.zip](https://github.com/OpenAPITools/openapi-generator/files/8168705/spec.yaml.zip) to reproduce the behaviour.

##### Steps to reproduce

This will fail: 
```
openapi-generator generate -i ~/Downloads/spec.yaml -g javascript -o ~/Downloads/output/
```

but by disabling the validation it will work 

```
openapi-generator generate -i ~/Downloads/spec.yaml -g javascript -o ~/Downloads/output/ --skip-validate-spec
```

**Comment by kain88-de** [comment]
Using the newest docker images (Digest: sha256:e6d68d4944d0a1cf904356290b3392c064241d508bbafea79b69e1340e4fa2d3) this is gone:
